### PR TITLE
Adding Running PU ProdLike Wfs

### PR DESCRIPTION
**Pull request by AdrianoDee**
#### PR description:

This PR proposes the addition of `ProdLike` workflows with running `PU` conditions (using `AVE_YY_BX_25ns` scenarios):

- from `10` to `100` for Run3;
- from `10` to `180` for Phase2.

These could be useful to cyclically test how the production timing scales in "continuous" PU conditions. The suffixes follows the scheme: `.21PU1` (given that `.21` is the suffix for `ProdLike` wfs). The missing `AVE_PU_BX_25ns` (and  `AVE_PU_BX_50ns` for consistency) scenarios have been added accordingly.
 
#### PR validation:

Running, e.g., `*.21101`, `*.21601` and `*.21801` wfs.

#### PR Backport

A back-port to `12_6_X` could be useful but not mandatory.

### CODE REVIEW

**Comment by AdrianoDee** [comment]
test parameters:

- workflows = 11634.1021, 11634.6021, 11634.4021, 11634.9021, 20834.16021, 20834.12021, 20834.18021, 20834.1021

**Comment by AdrianoDee** [comment]
please test

**Comment by AdrianoDee** [comment]
test parameters:

- workflows = 11634.1021, 11634.6021, 11634.4021, 11634.9021, 20834.16021, 20834.12021, 20834.18021, 20834.1021
- relvals_opt= -w upgrade

**Comment by AdrianoDee** [comment]
please test

**Comment by srimanob** [comment]
Hi @AdrianoDee 
Would it be easier with .21XXX for suffix? Will it look more like a same family of .21?
So we can keep the suffix, i.e 60, .70, .80, for other customizations.

**Comment by AdrianoDee** [comment]
Hi @srimanob, I adopted this because the trailing zeros are ignored (being a float) and then `.2110` and `.2100` would end up being duplicates.

**Comment by AdrianoDee** [comment]
Still, it could make sense to find a better way to avoid booking `.60‘ etc.

**Comment by kpedro88** [comment]
The numbering scheme is a little bit in conflict with the existing .9921, which is the PremixProdLike workflow. There's also a .999 workflow for Phase2 premix with PU 50, which is part of the short matrix. I'm not really that worried about the numbering scheme in general (since we don't tend to be too strict about it), but I just wanted to make sure everyone is aware of the existing workflows.

It would also be good to keep https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/README.md up to date...

**Comment by AdrianoDee** [comment]
@kpedro88, @srimanob thanks for the comments. I've updated the numbering scheme to be a bit more clear and unambiguous (I hope) to `.21PU1` with the trailing `1` to avoid the trailing zero problem. I've updated the README accordingly.

**Comment by AdrianoDee** [comment]
test parameters:

- workflows = 11634.21101, 11634.21601, 11634.21401, 11634.21901, 20834.211601, 20834.211201, 20834.211801, 20834.21101
- relvals_opt= -w upgrade

**Comment by AdrianoDee** [comment]
please test

**Comment by srimanob** [comment]
While reviewing this PR together with reco-profiling page of [Run-3](https://cms-reco-profiling.web.cern.ch/cms-reco-profiling/results/cmdlog/CMSSW_12_6_0_pre1/el8_amd64_gcc10/11834.21/cmdLog_profiling.txt) and [Phase-2](https://cms-reco-profiling.web.cern.ch/cms-reco-profiling/results/cmdlog/CMSSW_12_6_0_pre1/el8_amd64_gcc10/39634.21/cmdLog_profiling.txt), I see that we are doing something strange:
- For Run-3: We keep using FEVTDEBUG for GEN-SIM, while we update only Phase-2: https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py#L1372-L1373
- For Phase-2: We put back L1Reco step to RECO, https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py#L1378-L1379

I can make a fix later after this PR. 
FYI @cms-sw/reconstruction-l2

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -1422,6 +1422,136 @@ def condition(self, fragment, stepList, key, hasHarvest):
...
+            'MiniAOD',
+            'ALCA',
+            'ALCAPhase2',
+            'Nano',
+        ]):
+        super(UpgradeWorkflow_ProdLikeRunningPU, self).__init__(steps, PU, suffix, offset)
+        self.__fixedPU = fixedPU
+    def setupPU_(self, step, stepName, stepDict, k, properties):
+        #  change PU skipping ALCA and HARVEST
+        if not ('ALCA' in step or 'HARVEST' in step) and '--pileup' in stepDict[stepName][k]:
```
Comment (srimanob) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:1461]: Do we need to check ALCA and HARVEST here? I think it is removed already in ProdLike.

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 2]
```diff
@@ -1422,6 +1422,136 @@ def condition(self, fragment, stepList, key, hasHarvest):
     offset = 0.21,
 )
 
+class UpgradeWorkflow_ProdLikeRunningPU(UpgradeWorkflow_ProdLike):
+    def __init__(self, suffix, offset, fixedPU,
+         steps = [
+        'GenSimHLBeamSpot14',
```
Comment (srimanob) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:1428]: Why do we need all of these steps in `steps`? We don't need to treat NoPU in this customization.

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py (hunk 1 above)
Comment (AdrianoDee) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:1461]: If I remove the `ALCA`  and `HARVEST` checks I get:

```bash
  File "/home/adrianodif/running_pu/dev/bin/slc7_amd64_gcc11/runTheMatrix.py", line 622, in <module>
    ret = runSelected(opt)
  File "/home/adrianodif/running_pu/dev/bin/slc7_amd64_gcc11/runTheMatrix.py", line 31, in runSelected
    if len(undefSet)>0: raise ValueError('Undefined workflows: '+', '.join(map(str,list(undefSet))))
ValueError: Undefined workflows: 21034.21101
```

and this comes from the fact that `setup_` from `ProdLike` removes the step by setting `stepDict[stepName][k]=None`. So the steps are actually still there and I need to check if they are not `None`. I'm changing it to a more general:

```python
if stepDict[stepName][k] is not None and '--pileup' in stepDict[stepName][k]:
            stepDict[stepName][k]['--pileup'] = 'AVE_' + str(self.__fixedPU) + '_BX_25ns'
```

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py (hunk 2 above)
Comment (AdrianoDee) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:1428]: Indeed you are right I will remove them.

**Comment by srimanob** [comment]
@cmsbuild please test

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 3]
```diff
@@ -1422,6 +1422,122 @@ def condition(self, fragment, stepList, key, hasHarvest):
...
+# trailing zeros are ignored. Thus 0.21N1 is used
+
+upgradeWFs['ProdLikePU10'] = UpgradeWorkflow_ProdLikeRunningPU(
+    suffix = '_ProdLikePU10',
+    offset = 0.21101,
+    fixedPU = 10,
+)
+
+upgradeWFs['ProdLikePU20'] = UpgradeWorkflow_ProdLikeRunningPU(
+    suffix = '_ProdLikePU10',
```
Comment (srimanob) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:1464]: Seem to be a typo here, sorry I just see it.

Comment (AdrianoDee) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py]: Right! Fixing it, thanks

**Comment by AdrianoDee** [comment]
please test

**Comment by AdrianoDee** [comment]
please test

**Comment by srimanob** [comment]
+Upgrade

The failure in comparison of 11634.7 does not look like coming from this PR.

**Comment by srimanob** [comment]
Kindly ping @cms-sw/pdmv-l2 
Thx.

**Comment by kskovpen** [comment]
+pdmv

**Comment by AdrianoDee** [comment]
@cms-sw/orp-l2 just a ping because this is basically fully signed (while it doesn't appear so for the `operations-pending`). Thanks.

**Comment by perrotta** [comment]
+1

**Pull request merged**